source/cppfs/source/posix/LocalFileWatcher.cpp

```diff
@@ -0,0 +1,147 @@
+
+#include <cppfs/posix/LocalFileWatcher.h>
+
+#include <unistd.h>
+#include <limits.h>
+#include <sys/inotify.h>
+
+#include <cppfs/cppfs.h>
+#include <cppfs/FilePath.h>
+#include <cppfs/FileHandle.h>
+#include <cppfs/FileIterator.h>
+#include <cppfs/FileWatcher.h>
+#include <cppfs/posix/LocalFileSystem.h>
+#include <cppfs/posix/LocalFileIterator.h>
+
+
+namespace cppfs
+{
+
+
+LocalFileWatcher::LocalFileWatcher(FileWatcher * fileWatcher, std::shared_ptr<LocalFileSystem> fs)
+: AbstractFileWatcherBackend(fileWatcher)
+, m_fs(fs)
+, m_inotify(-1)
+{
+    // Create inotify instance
+    m_inotify = inotify_init();
+}
+
+LocalFileWatcher::~LocalFileWatcher()
+{
+    // Close watch handles
+    for (auto it : m_watchers) {
+        inotify_rm_watch(m_inotify, it.first);
+    }
+
+    // Close inotify instance
+    close(m_inotify);
+}
+
+AbstractFileSystem * LocalFileWatcher::fs() const
+{
+    // Return file system
+    return static_cast<AbstractFileSystem *>(m_fs.get());
+}
+
+void LocalFileWatcher::add(FileHandle & fh, unsigned int events, RecursiveMode recursive)
+{
+    // Get watch mode
+    uint32_t flags = 0;
+    if (events & FileCreated)  flags |= IN_CREATE;
+    if (events & FileRemoved)  flags |= IN_DELETE;
+    if (events & FileModified) flags |= (IN_MODIFY | IN_ATTRIB);
```

Comment:
I think we should add a new type FileAttribChanged instead of implicitly merging them.